mailmerge/__main__.py

```diff
@@ -112,7 +119,40 @@ def main(sample, dry_run, limit, no_limit, resume,
         template_message = TemplateMessage(template_path)
         csv_database = read_csv_database(database_path)
         sendmail_client = SendmailClient(config_path, dry_run)
+
+        # Setup rate limit if needed
```

Comment:
I think the right abstraction to modify is `SendmailClient`.  Perhaps adding a new parameter to its `__init__()` function with the rate limit and a member variable to remember the timestamp of the last message sent would work.

Reply:
Makes sense, but that also requires a return code for rate limit exceeded and code to handle the waiting in main().

Reply:
How about if `SendmailClient::sendmail()` throws an exception if the rate limit is exceeded?  It could be a custom exception type that inherits from `MailmergeError`, maybe `MailmergeRateLimitError`.  You could even include a suggested wait time.

An alternative: the sendmail client could have an input that's a callback function.  The client could call the callback every second to update the output.  Main would provide a function something like `lambda: print(".")` as an input to the `SendmailClient::__init__()` function.

Reply:
Addressed in the latest commits.